test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,386 @@
+		ginkgo.Fail("Networking settings are missing or incomplete")
+	}
+	if StringSet(cfg.DeployGardenletOnShoot) {
+		if _, err := strconv.ParseBool(cfg.DeployGardenletOnShoot); err != nil {
+			ginkgo.Fail("-deploy-gardenlet is not a boolean value")
+		}
+	} else {
+		cfg.DeployGardenletOnShoot = "true"
+	}
+}
+
+func mergeSeedConfig(base, overwrite *SeedRegistrationConfig) *SeedRegistrationConfig {
+	if base == nil {
+		return overwrite
+	}
+	if overwrite == nil {
+		return base
+	}
+	if overwrite.GardenerConfig != nil {
+		base.GardenerConfig = overwrite.GardenerConfig
+	}
+	if StringSet(overwrite.SeedName) {
+		base.SeedName = overwrite.SeedName
+	}
+	if StringSet(overwrite.SecretBinding) {
+		base.SecretBinding = overwrite.SecretBinding
+	}
+	if StringSet(overwrite.BackupSecretProvider) {
+		base.BackupSecretProvider = overwrite.BackupSecretProvider
+	}
+	if StringSet(overwrite.ShootedSeedName) {
+		base.ShootedSeedName = overwrite.ShootedSeedName
+	}
+	if StringSet(overwrite.ShootedSeedNamespace) {
+		base.ShootedSeedNamespace = overwrite.ShootedSeedNamespace
+	}
+	if StringSet(overwrite.ShootedSeedKubeconfig) {
+		base.ShootedSeedKubeconfig = overwrite.ShootedSeedKubeconfig
+	}
+	if StringSet(overwrite.ProviderRegion) {
+		base.ProviderRegion = overwrite.ProviderRegion
+	}
+	if StringSet(overwrite.ProviderType) {
+		base.ProviderType = overwrite.ProviderType
+	}
+	if StringSet(overwrite.NetworkingBlockCIDRs) {
+		base.NetworkingBlockCIDRs = overwrite.NetworkingBlockCIDRs
+	}
+	if StringSet(overwrite.NetworkingPods) {
+		base.NetworkingPods = overwrite.NetworkingPods
+	}
+	if StringSet(overwrite.NetworkingServices) {
+		base.NetworkingServices = overwrite.NetworkingServices
+	}
+	if StringSet(overwrite.NetworkingNodes) {
+		base.NetworkingNodes = overwrite.NetworkingNodes
+	}
+	if StringSet(overwrite.DeployGardenletOnShoot) {
+		base.DeployGardenletOnShoot = overwrite.DeployGardenletOnShoot
+	}
+	return base
+}
+
+// RegisterSeedRegistrationFrameworkFlags adds all flags that are needed to configure a seed registration framework to the provided flagset.
+func RegisterSeedRegistrationFrameworkFlags() *SeedRegistrationConfig {
+	_ = RegisterGardenerFrameworkFlags()
+
+	newCfg := &SeedRegistrationConfig{}
+
+	flag.StringVar(&newCfg.SeedName, "seed-name", "", "Name of the seed")
+	flag.StringVar(&newCfg.SecretBinding, "secret-binding", "", "Secret binding parameter")
+	flag.StringVar(&newCfg.BackupSecretProvider, "backup-secret-provider", "", "Provider of the backup secret reference")
+	// Shooted seed reference
+	flag.StringVar(&newCfg.ShootedSeedName, "shoot-name", "", "Name of the shoot")
+	flag.StringVar(&newCfg.ShootedSeedNamespace, "shoot-namespace", "", "Namespace of the shoot")
+	flag.StringVar(&newCfg.ShootedSeedKubeconfig, "shoot-kubecfg", "", "Path to the kubeconfig of the shooted seed")
+	// Provider
+	flag.StringVar(&newCfg.ProviderType, "provider-type", "", "Provider type e.g. aws, az, gcp")
```

Comment:
```suggestion
	flag.StringVar(&newCfg.ProviderType, "provider-type", "", "Provider type e.g. aws, azure, gcp")
```

Ref https://github.com/gardener/gardener-extension-provider-azure/blob/v1.14.0/example/controller-registration.yaml#L9